### ISSUE TITLE
[ECS] [request]: Customize Service Connect configuration

### ISSUE THREAD

**Issue by Hugoz** [aws/containers-roadmap#1963]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Allow customization of Service Connect configuration related to retries and timeout per target

**Which service(s) is this request for?**
ECS Service Connect

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Today SC will retry twice on connection failure or 503 responses. Customize this policy based on the target service would be great. For example disable retries for requests to a "Process Payment" service that is not idempotent. 
When Authentication is supported, this could be extended to define which services require Auth and which don't.

**Are you currently working around this issue?**
Don't use Service Connect

**Comment by herrhound** [comment]
@Hugoz – In the example that you've provided you're talking about disabling retries to a payment processor. Leaving the topic of non-idempotent payment processors aside, do you expect a service that returns a 503 error to actually accept the request?

**Comment by Hugoz** [comment]
Yes 503 could have many causes depending on the programming language. A service could process the request and just have a bug in the response builder.

**Comment by juanda-agero** [comment]
Is there something on the roadmap to be able to handle TLS termination on the sidecar container? It seems like the current issue is related to retries and timeouts only, but it would be very helpful to be able to also enable TLS handling on the Envoy container. 

As per recent support session with an AWS engineer, it's impossible to handle TLS on the sidecar container and TLS must be handled by the application itself, which means dealing with certificate generation, expiration, etc.

Is there something similar on the roadmap? We need to enable TLS between services in the same namespace due to possible PII flowing between services in the cluster.

**Comment by atalwar-sephora** [comment]
Agree with @Hugoz We are also looking for similar functionality and control in service connect configuration. This will also be handy in situations where the upstream service is still coming up while the container is registered in CloudMap and service connect will start sending request to upstream and receive 5xx since the application is not ready yet. 
Having control retries and timeout settings for upstream will help in this situation.
